RecoPixelVertexing/PixelTrackFitting/test/PixelTrackRiemannFit.cc

```diff
@@ -0,0 +1,427 @@
+  Vector8d z_err;
+  Vector6d hand;
+  Vector3i hand_2;
+  Vector6d xy_err;
+  Vector6d zh_err;
+  double z_max;
+  double r_max;
+};
+
+void test_helix_fit();
+
+constexpr int c_speed = 299792458;
+constexpr double pi = M_PI;
+default_random_engine generator(1);
+
+void smearing(const Vector5d& err, const bool& isbarrel, double& x, double& y, double& z) {
+  normal_distribution<double> dist_R(0., err[0]);
+  normal_distribution<double> dist_Rp(0., err[1]);
+  normal_distribution<double> dist_z(0., err[2]);
+  normal_distribution<double> dist_xyh(0., err[3]);
+  normal_distribution<double> dist_zh(0., err[4]);
+  if (isbarrel) {
+    double dev_Rp = dist_Rp(generator);
+    double dev_R = dist_R(generator);
+    double R = sqrt(Rfit::sqr(x) + Rfit::sqr(y));
+    x += dev_Rp * +y / R + dev_R * -x / R;
+    y += dev_Rp * -x / R + dev_R * -y / R;
+    z += dist_z(generator);
+  } else {
+    x += dist_xyh(generator);
+    y += dist_xyh(generator);
+    z += dist_zh(generator);
+  }
+}
+
+template <int N>
+void Hits_cov(Eigen::Matrix<float, 6, 4>& V,
+              const unsigned int& i,
+              const unsigned int& n,
+              const Matrix3xNd<N>& hits,
+              const Vector5d& err,
+              bool isbarrel) {
+  if (isbarrel) {
+    double R2 = Rfit::sqr(hits(0, i)) + Rfit::sqr(hits(1, i));
+    V.col(i)[0] = (Rfit::sqr(err[1]) * Rfit::sqr(hits(1, i)) + Rfit::sqr(err[0]) * Rfit::sqr(hits(0, i))) / R2;
+    V.col(i)[2] = (Rfit::sqr(err[1]) * Rfit::sqr(hits(0, i)) + Rfit::sqr(err[0]) * Rfit::sqr(hits(1, i))) / R2;
+    V.col(i)[1] = (Rfit::sqr(err[0]) - Rfit::sqr(err[1])) * hits(1, i) * hits(0, i) / R2;
+    V.col(i)[5] = Rfit::sqr(err[2]);
+  } else {
+    V.col(i)[0] = Rfit::sqr(err[3]);
+    V.col(i)[2] = Rfit::sqr(err[3]);
+    V.col(i)[5] = Rfit::sqr(err[4]);
+  }
+}
+
+hits_gen Hits_gen(const unsigned int& n, const Matrix<double, 6, 1>& gen_par) {
+  hits_gen gen;
+  gen.hits = MatrixXd::Zero(3, n);
+  gen.hits_ge = Eigen::Matrix<float, 6, 4>::Zero();
+  // err /= 10000.;
+  constexpr double rad[8] = {2.95, 6.8, 10.9, 16., 3.1, 7., 11., 16.2};
+  // constexpr double R_err[8] = {5./10000, 5./10000, 5./10000, 5./10000, 5./10000,
+  // 5./10000, 5./10000, 5./10000};  constexpr double Rp_err[8] = {35./10000, 18./10000,
+  // 15./10000, 34./10000, 35./10000, 18./10000, 15./10000, 34./10000};  constexpr double z_err[8] =
+  // {72./10000, 38./10000, 25./10000, 56./10000, 72./10000, 38./10000, 25./10000, 56./10000};
+  constexpr double R_err[8] = {
+      10. / 10000, 10. / 10000, 10. / 10000, 10. / 10000, 10. / 10000, 10. / 10000, 10. / 10000, 10. / 10000};
+  constexpr double Rp_err[8] = {
+      35. / 10000, 18. / 10000, 15. / 10000, 34. / 10000, 35. / 10000, 18. / 10000, 15. / 10000, 34. / 10000};
+  constexpr double z_err[8] = {
+      72. / 10000, 38. / 10000, 25. / 10000, 56. / 10000, 72. / 10000, 38. / 10000, 25. / 10000, 56. / 10000};
+  const double x2 = gen_par(0) + gen_par(4) * cos(gen_par(3) * pi / 180);
```

Comment:
`pi` or M_PI as in the https://github.com/cms-sw/cmssw/blob/b7a0e939f81a5bbcba86c6ded118a46e5f2f7f50/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h#L131 ?

Reply:
I think it's the same: `constexpr double pi = M_PI;` though it can be simplified of course